lib/progress.js

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+import { CodeExample, PropsDefinition, Prop } from './story-utils';
+
+export const Progress = styled.progress.attrs(() => ({ 'data-module': 'Progress' }))`
+  appearance: none;
+  display: block;
+  height: 0.75em;
+  width: 100%;
+  border: 1px solid currentColor;
+  background-color: transparent; // firefox
+  border-radius: var(--rounded);
+
+  // ugly prefixes sadly required
+  &::-webkit-progress-bar {
+    background-color: transparent;
+  }
+  &::-webkit-progress-value {
+    background-color: currentColor;
+    border-radius: calc(var(--rounded) - 2px);
+  }
+  &::-moz-progress-bar {
+    background-color: currentColor;
+    border-radius: calc(var(--rounded) - 2px);
+  }
+`;
+Progress.propTypes = {
+  max: PropTypes.number.isRequired,
+  value: PropTypes.number.isRequired,
+  children: PropTypes.node.isRequired,
+};
+
+export const StoryProgress = () => (
+  <>
+    <p>
+      The Progress component renders a styled <code>{`<progress>`}</code> element that matches the surrounding text color and background.
+    </p>
+    <CodeExample>{`<Progress max={100} value={30}>30%</Progress>`}</CodeExample>
+    <PropsDefinition>
+      <Prop name="value" required>
+        A number between 0 and the value of "max" that indicates how much of the task has been completed.
+      </Prop>
+      <Prop name="max" required>
+        The maxiumum value of the progress indicator.
+      </Prop>
+    </PropsDefinition>
+    <div style={{ width: '10rem' }}>
+      <div style={{ backgroundColor: 'var(--colors-secondaryBackground)', padding: 'var(--space-1)' }}>
+        <p>Secondary background styles</p>
+        <Progress value={30} max={100}>
+          30%
+        </Progress>
+      </div>
+      <div style={{ color: 'var(--colors-notice-text)', backgroundColor: 'var(--colors-notice-background)', padding: 'var(--space-1)' }}>
+        <p>Notification styles</p>
+        <Progress value={70} max={100}>
```

Comment:
Voiceover is reading this as "indeterminate progress indicator" instead of "70% progress indicator" like it should be. I copied the markup (as well as some variations on the markup) and I hear the correct behavior in https://nebulous-crab.glitch.me/. 

Without doing more digging I'm kinda stumped on why this isn't working - I'm testing both in Chrome on Mac, in the same browser window.

Reply:
Found that someone else has experienced the same thing testing styled progress bars in Chrome with Voiceover. [see link](https://scottaohara.github.io/a11y_styled_form_controls/src/progress-bar/). This would explain why my demo app works - it's using unstyled progress bars. 

> Though the progress element is largely reporting the correct information to browsers, screen readers have quite a few kinks in how they announce the element to users.
> 
> At the time of testing, a styled progress bar won't be fully accessible in all screen reader and browser pairings. Instead it may be more appropriate to simply treat the progress bar as a visual decoration, hide it from screen readers, and provide visually hidden text as a means to consistently convey the information.

@modernserf what do you think about modifying the component to take this approach? Using the VisuallyHidden component for the text-only version and aria-hidden='true' for the `<progress>` element should accomplish what we need without much extra work/markup.